docs/filesystem.rst

```diff
@@ -0,0 +1,227 @@
+    configured instance of this class is returned by the ``open`` function
+    described above.
+
+    .. py:method:: close()
+
+        Flush and close the file. This method has no effect if the file is
+        already closed. Once the file is closed, any operation on the file
+        (e.g. reading or writing) will raise an exception.
+
+    .. py:method:: name()
+
+        Returns the name of the file the object represents. This will be the
+        same as the ``filename`` argument passed into the call to the ``open``
+        function that instantiated the object.
+
+    .. py:method:: read(size)
+
+         Read up to ``size`` bytes from the object and return them. As a
+         convenience, if ``size`` is unspecified or -1, ``readall()`` is
+         called. Fewer than ``size`` bytes may be returned if there are less
+         than ``size`` bytes remaining to be read from the file.
+
+         If 0 bytes are returned, and ``size`` was not 0, this indicates end of
+         file.
+
+    .. py:method:: readall()
+
+        Read and return all the bytes from the stream until EOF (end of file)
+        is reached.
+
+    .. py:method:: readinto(buf, n=-1)
+
+        Read bytes into the buffer ``buf``. If ``n`` is supplied, read ``n``
+        number of bytes into the buffer ``buf``.
+
+    .. py:method:: readline(size)
+
+        Read and return one line from the file. If ``size`` is specified, at
+        most ``size`` bytes will be read.
+
+        The line terminator is always ``b'\n'`` for binary files.
+
+    .. py:method:: writable()
+
+        Return ``True`` if the file supports writing. If ``False``, ``write()``
+        will raise ``OSError``.
+
+    .. py:method:: write(buf)
+
+        Write the bytes in buffer ``buf`` to the file and return the number of
+        bytes written.
+
+.. py:module:: os
+
+MicroPython contains an ``os`` module based upon the ``os`` module in the
+Python standard library. It's used for accessing what would traditionally be
+termed as operating system dependent functionality. Since there is no operating
+system in MicroPython the module provides functions relating to the management
+of the simple on-device persistent file system and information about the
+current system.
+
+To access this module you need to::
+
+    import os
+
+We assume you have done this for the examples below.
+
+Functions
+=========
+
+.. py:function:: listdir()
+
+    Returns a list of the names of all the files contained within the local
+    persistent on-device file system.
+
+.. py:function:: remove(filename)
+
+    Removes (deletes) the file named in the argument ``filename``. If the file
+    does not exist an ``OSError`` exception will occur.
+
+.. py:function:: size(filename)
+
+    Returns the size, in bytes, of the file named in the argument ``filename``.
+    If the file does not exist an ``OSError`` exception will occur.
+
+.. py:function:: uname()
```

Comment:
Well, given that these functions sit under the ``.. py:module:: os`` declaration in RST it's logically the right place. Totally get how this might be seen as an odd fit, but I'm not sure where else it would go.

Suggestions?